apps/arweave/src/ar_nonce_limiter_server_worker.erl

```diff
@@ -36,46 +39,43 @@ start_link(Name, RawPeer) ->
 init(RawPeer) ->
 	process_flag(trap_exit, true),
 	ok = ar_events:subscribe(nonce_limiter),
-	gen_server:cast(self(), resolve_raw_peer),
+	case ar_config:is_public_vdf_server() of
+		false ->
+			gen_server:cast(self(), re_resolve_peer_domain);
+		true ->
+			ok
+	end,
 	{ok, #state{ raw_peer = RawPeer }}.
 
 handle_call(Request, _From, State) ->
 	?LOG_WARNING([{event, unhandled_call}, {module, ?MODULE}, {request, Request}]),
 	{reply, ok, State}.
 
-handle_cast(resolve_raw_peer, #state{ raw_peer = RawPeer } = State) ->
+handle_cast(re_resolve_peer_domain, #state{ raw_peer = RawPeer } = State) ->
 	case ar_peers:resolve_and_cache_peer(RawPeer, vdf_client_peer) of
-		{ok, Peer} ->
-			ar_util:cast_after(?RESOLVE_DOMAIN_NAME_FREQUENCY_MS, self(), resolve_raw_peer),
-			{noreply, State#state{ peer = Peer }};
-		{error, Reason} ->
-			?LOG_WARNING([{event, failed_to_resolve_vdf_client_peer},
-					{peer, RawPeer},
-					{reason, io_lib:format("~p", [Reason])}]),
-			ar_util:cast_after(10000, self(), resolve_raw_peer),
-			{noreply, State}
-	end;
+		{ok, _} ->
+			ok;
+		Error ->
+			?LOG_WARNING([{event, failed_to_re_resolve_peer_domain},
+					{error, io_lib:format("~p", [Error])},
+					{peer, io_lib:format("~p", [RawPeer])}])
+	end,
+	ar_util:cast_after(?RE_RESOLVE_PEER_DOMAIN_MS, ?MODULE, re_resolve_peer_domain),
+	{noreply, State};
 
 handle_cast(Cast, State) ->
 	?LOG_WARNING([{event, unhandled_cast}, {module, ?MODULE}, {cast, Cast}]),
 	{noreply, State}.
 
-handle_info({event, nonce_limiter, _Event}, #state{ peer = undefined } = State) ->
-	{noreply, State};
 handle_info({event, nonce_limiter, {computed_output, Args}}, State) ->
-	#state{ peer = Peer, pause_until = Timestamp, format = Format } = State,
-	{SessionKey, StepNumber, Output, _PartitionUpperBound} = Args,
-	CurrentStepNumber = ar_nonce_limiter:get_current_step_number(),
-	case os:system_time(second) < Timestamp of
-		true ->
+	#state{ raw_peer = RawPeer } = State,
+	case ar_peers:resolve_and_cache_peer(RawPeer, vdf_client_peer) of
```

Comment:
since we always check the cache and re-resolve here, can we remove the periodic `re_resolve_peer_domain`?